packages/nodes-base/nodes/KoboToolbox/descriptions/Operations.ts

```diff
@@ -0,0 +1,130 @@
+import {
+	INodeProperties,
+} from 'n8n-workflow';
+
+export const operations = [
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				displayOptions: {
+					show: {
+						resource: [
+							'form',
+						],
+					},
+				},
+				options: [
+					{
+						name: 'Get',
+						value: 'get',
+						description: 'Retrieve a form definition',
+					},
+				],
+				default: 'get',
+				description: 'The operation to perform.',
+			},
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				displayOptions: {
+					show: {
+						resource: [
+							'submission',
+						],
+					},
+				},
+				options: [
+					{
+						name: 'Query',
+						value: 'query',
+						description: 'Query matching submissions',
+					},
+					{
+						name: 'Get',
+						value: 'get',
+						description: 'Get a single submission',
+					},
+					{
+						name: 'Delete',
+						value: 'delete',
+						description: 'Delete a single submission',
+					},
+					{
+						name: 'Get Validation Status',
+						value: 'get_validation',
+						description: 'Get the validation status for the submission',
+					},
+					{
+						name: 'Update Validation Status',
+						value: 'set_validation',
+						description: 'Set the validation status of the submission',
+					},
+				],
+				default: 'query',
+				description: 'The operation to perform.',
+			},
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				displayOptions: {
+					show: {
+						resource: [
+							'attachment',
```

Comment:
You're right, the 'download' feature is really only an enrichment of the operations that retrieve a form submission.
I've removed this resource/operation and moved it as an option in the `submission` `get` and `query` operations!

Reply:
Note that Kobo apparently changed recently the media download flow. It now includes an HTTP redirect, which breaks the authentication (since redirects won't carry over the original request headers).
I had to change this implementation a bit, and needed to use axios directly - the wrappers won't let me handle the redirect myself since they will throw on receiving a 301.